### DISCUSSION TITLE
Updating UI from Background Thread stops working after Navigating to a Page more than once

### DISCUSSION THREAD

**Discussion by ebondur** [unoplatform/uno#18178] [Q&A] [answered]
I am currently trying to update the UI of a page when an NFC read occurs. As monitoring the android device for NFC messages occurs on a background thread, the UI is updated from the background thread as well. When I first navigate to the page, everything works as expected. However, when I navigate away and back to the page, all UI updating stops. Here is the current behavior using a button to simulate NFC reads:

https://github.com/user-attachments/assets/e8bd98c2-4a74-4938-b0ea-a655024d47de

Expected behavior is that The UI should update every time if the updates originate from the background thread, even if the page has been navigated to more than once. Only way to fix the issue is to close and reopen the app, which is not viable solution.

[Sample Repo](https://github.com/ebondur/UnoBackgroundUIUpdateBug) can be found here.

**Comment by MartinZikmund** [comment, marked as answer]
Created https://github.com/unoplatform/uno/issues/18182 to track this
